images/hub/jupyterhub_config.py

```diff
@@ -285,6 +285,18 @@ def generate_user_name(spawner):
         'affinity': {
             'podAffinity': {
                 'preferredDuringSchedulingIgnoredDuringExecution': [{
+                    'weight': 1000,
+                    'podAffinityTerm': {
+                        'labelSelector': {
+                            'matchExpressions': [{
+                                'key': 'component',
+                                'operator': 'In',
+                                'values': ['hub']
```

Comment:
do these match regular expression-style? We should match "startswith" hub if possible, in case a user creates a pod w/ the word "hub" in it

Reply:
Good question, I have presumed not, but i'll find out.

Reply:
`matchLabels` is a map of `{key,value}` pairs. A single `{key,value}` in the `matchLabels` map is equivalent to an element of `matchExpressions`, whose `key` field is “key”, the `operator` is “In”, and the `values` array contains only “value”. 

--- https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ ---

In other words, matchExpressions with key: component, operator: In, values: [hub] is equal to matchLabels: [component:hub], and that probably is an exact match rather than wildcarded like *hub* or similar.

---

The matchExpressions naming is probably referring to that the operator can vary, it can be `In`, `NotIn` or `Exist` expressions. They only seem to support equality or inequality.

Reply:
When I read the k8s docs, I recall an impression that one was encouraged to use matchExpressions rather than matchLabels. Perhaps just confirmation bias by now, but they could do all the things matchLabels can do and more.

Reply:
TL;DR; They match with equality/inequality rather than regex matching/not matching.